enigma-principal/app/src/esgx/keys_keeper_u.rs

```
@@ -44,7 +45,7 @@ pub fn get_enc_state_keys(eid: sgx_enclave_id_t, request: StateKeyRequest, epoch
         )
     };
     if retval != EnclaveReturn::Success || status != sgx_status_t::SGX_SUCCESS {
-        return Err(EnclaveFailError { err: retval, status }.into());
+        bail!("{:?} encountered in the enclave", retval);
```

Same as above

---

You're right. I should have put a TODO here. It was swallowing the exception so I did this temporarily. I'll take a second look and do it properly.

---

Fixed in commit 3f28b0a0cf3d1edf20ec81249b4dcd56bb3b79d0. I'll add some negative test cases to make sure that some trace of the enclave error gets carried over.